### PR TITLE
Update TEveMu2eDataInterface.cc

### PR DESCRIPTION
**Pull request by NamithaChitrazee**
Includes KalSegment information which is used to display the reconstructed track

### CODE REVIEW

File: TEveEventDisplay/src/TEveMu2eDataInterface.cc [hunk 1]
```diff
@@ -376,43 +377,59 @@ template <typename L> void maxminCRV(L data, double &max, double &min){
...
+      line->fKalSeed = kseed;
+      line->SetSeedInfo(kseed);
+  
+      for(size_t m=0; m<nSegments; m++)
+      {
+        const mu2e::KalSegment &segment = segments.at(m);
+
+        fltLMin=segment.fmin();
+        fltLMax=segment.fmax();
+        std::cout<<"Inside loop , fmin ="<<fltLMin<<" fmax="<<fltLMax<<std::endl;
```
Comment (brownd1978) [inline TEveEventDisplay/src/TEveMu2eDataInterface.cc:405]: this looks like a debugging statement.  It should be put in an if(debug_){   clause.

Comment (NamithaChitrazee) [reply TEveEventDisplay/src/TEveMu2eDataInterface.cc]: Yes, thank you. I think I have taken care of it now.

**Review by sophiemiddleton — APPROVED** [review]
Following on from Dave's Review all this looks great Namitha - well done!

File: TEveEventDisplay/src/TEveMu2eDataInterface.cc [hunk 2]
```diff
@@ -376,43 +377,54 @@ template <typename L> void maxminCRV(L data, double &max, double &min){
...
+        for(unsigned int k = 0; k < seedcol->size(); k = k + 20){   
+       
           KalSeed kseed = (*seedcol)[k];
-          TEveMu2eCustomHelix *line = new TEveMu2eCustomHelix();
-          TEveMu2eCustomHelix *line_twoD = new TEveMu2eCustomHelix();
-          line->fKalSeed = kseed;
-          line->SetSeedInfo(kseed);
+          const std::vector<mu2e::KalSegment> &segments = kseed.segments();
+      size_t nSegments=segments.size();
+      if(nSegments==0) continue;
```
Comment (kutschke) [inline TEveEventDisplay/src/TEveMu2eDataInterface.cc:385]: Please follow the indentation patterns established by the earlier developers.

File: TEveEventDisplay/src/TEveMu2eDataInterface.cc [hunk 3]
```diff
@@ -376,43 +377,54 @@ template <typename L> void maxminCRV(L data, double &max, double &min){
...
+      const mu2e::KalSegment &segmentFirst = kseed.segments().front();
+      const mu2e::KalSegment &segmentLast = kseed.segments().back();
+      double fltLMin=segmentFirst.fmin();
+      double fltLMax=segmentLast.fmax();
+    
+      TEveMu2eCustomHelix *line = new TEveMu2eCustomHelix();
+      TEveMu2eCustomHelix *line_twoD = new TEveMu2eCustomHelix();
```
Comment (kutschke) [inline TEveEventDisplay/src/TEveMu2eDataInterface.cc:392]: This is for Sophie:  I see that the line and line_twoD objects are passed to collections of type TEveElementList.  Does that class take ownership and look after deletion?

**Comment by kutschke** [comment]
Sophie: while looking to where the pointees of line and line_twoD were deleted, I noticed the line:

https://github.com/NamithaChitrazee/Offline/blob/d0997c4468ac2bd511538f1bea2bb04bd85f4c72/TEveEventDisplay/src/TEveMu2eDataInterface.cc#L415

I have several comments that I would like you to address as targets of opportunity.   And keep your eyes open for other places in the code where you do the same thing.

This code will not draw that track all of the way to fltLMax - it will stop when it gets close.   I think it could be confusing to end users who are zooming in if the track stops before they expect it to.    Two other comments: you should never use floating point types as control variables in loops - you need to manage the possibility of round off errors.  Second please don't use literal numerics ( the increment unit of 1.0 ) - instead define a constexpr .  Can you remind me if the base unit of length in TEve event display is mm or cm or ???

**Comment by kutschke** [comment]
@FNALbuild run build tests

File: TEveEventDisplay/src/TEveMu2eDataInterface.cc (hunk 3 above)
Comment (sophiemiddleton) [reply TEveEventDisplay/src/TEveMu2eDataInterface.cc:392]: Ah, good point, I've never officially checked but assumed once out of scope it is gone....I need to spend some time debugging anyway. Once Namitha's work is merged I'll run some checks

**Comment by kutschke** [comment]
Thanks Sophie,

    Once she fixes the indentation, I will merge.  Please systematically check that everything you “new” gets deleted somewhere.  Hopefully the TEve documentation will be clear and you won’t need to read their code.

Rob


@sophiemiddleton commented on this pull request.

________________________________

In TEveEventDisplay/src/TEveMu2eDataInterface.cc<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_442-23discussion-5Fr629690554&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=wVM2mTxMhawakoas2lSomgsps6E56MsfaIS86OMbrcI&s=3HUnOoK7fa70y2opGjRsNffUA6sTZI9EI8zYAn9Z3ZI&e=>:

Ah, good point, I've never officially checked but assumed once out of scope it is gone....I need to spend some time debugging anyway. Once Namitha's work is merged I'll run some checks

**Comment by NamithaChitrazee** [comment]
> Following on from Dave's Review all this looks great Namitha - well done!

Thank you Sophie! I will check the code again and send it you for another review.

**Comment by kutschke** [comment]
@FNALbuild run build tests

**Pull request merged**